cmd/thanos/receive.go

```diff
@@ -225,6 +250,12 @@ func runReceive(
 			},
 		)
 	}
+
+	err := addShipper(logger, reg, g, metadata.ReceiveSource, &promMetadata{labels: externalLabels}, dataDir, objStoreConfig, false)
+	if err != nil {
+		return errors.Wrap(err, "failed to add block shipper to run group")
```

Comment:
Minor nit, when you wrap error for Thanos, let's avoid `failed to` prefixes. It's already known it failed. Wrap is only to figure out where it fail and why NOT to check IF it failed (: 
```suggestion
		return errors.Wrap(err, "add block shipper to run group")
```